File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        output = []

        if not nums:
            return []
        
        curVal=1
        for i in range(len(nums)):
            for j in range(len(nums)):
                if i != j:
                    curVal *= nums[j]            
            output.append(curVal)
            curVal = 1
        return output  

        

File: Data Structures & Algorithms/special-array-i/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    isArraySpecial(nums) {
        if (nums.length == 1) return true;
        let j = 0;  
        for (let i = 1; i < nums.length; i++) {
            if (
                (nums[i] % 2 == 0 && nums[j] % 2 == 0) ||
                (nums[i] % 2 !== 0 && nums[j] % 2 !== 0)
            ) {
                return false;
            }
            j++;
        }
        return true;
    }
}
